### PR TITLE
Update TiDB and TiKV configuration to 4.0.0-rc.2

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Close https://github.com/pingcap/tidb-operator/issues/2467
Close https://github.com/pingcap/tidb-operator/issues/1661

### Does this PR introduce a user-facing change?:

```release-note
Support TiDB and TiKV Configuration to v4.0.0-rc.2
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tiflash_config.go [hunk 1]
```diff
@@ -32,9 +32,9 @@ type TiFlashConfig struct {
 // +k8s:openapi-gen=false
 type FlashServerConfig struct {
 	// +optional
-	EngineAddr string `json:"engine-addr,omitempty" toml:"engine-addr,omitempty"`
+	EngineAddr *string `json:"engine-addr,omitempty" toml:"engine-addr,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tiflash_config.go:35]: Please also update the default config logic in https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tiflash_util.go#L122.
And also make sure that TiFlash can start up successfully, it requires some configs to exist even if they are empty strings.

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tiflash_config.go:35]: updated.

**Pull request merged**